src/main/java/com/shibumi/MaximumPathSum.java

```diff
@@ -10,6 +11,36 @@ public MaximumPathSum(int[][] matrix) {
         this.matrix = matrix;
     }
 
+    public String findMaximumPath() {
+
+        TreeMap<Integer, String> paths = new TreeMap<>();
+        accumulateAllPaths(0, 0, "", 0, paths);
+        return paths.lastEntry().getValue().trim();
+    }
+
+    public void accumulateAllPaths(int row, int col, String path, int total, TreeMap<Integer, String> paths) {
```

Comment:
2 questions:

1. Can you explain the purpose of the "paths" argument?
2. It looks like this is the brute-force way to solve the problem. Is there a way to adapt the algorithm from calculateMaxSum, without resorting to brute-force?

Reply:
1. The purpose of the TreeMap was to store the paths by key (the total) and the value (the path) and then use the last entry in that ordered map to get the element that corresponds to the max path. 
2. Nothings comes to mind right now but I'll think about this as soon as I get home. The main problem I had was that in the calculateMaxSum approach the "max value" is calculated for all the cells and the last cell will always end up with the higher sum but that didn't necessary indicated a path, I'll review the code again and see how I can adapt it to do so.

Reply:
I removed the method that was using the brute force approach, did some refactoring and use the memo grid to find the best path backwards.